### DISCUSSION TITLE
Broker configuration for redelivery of messages

### DISCUSSION THREAD

**Discussion by balkrishan333** [apache/pulsar#19042] [Q&A]
Is there some broker configuration which can help in redeliver the messages if messages are not acknowledged in specific time. I know about the Consumer API using which I can set ackTimeout but I am looking for some configuration on broker so that it can be applied to all consumers and topics.

**Comment by eolivelli** [comment]
Until the consumer disconnects or sends a NACK the broker thinks that the message is owned by it.
Why would you want to redeliver the message?

I think that we need more a mechanism to disconnect consumers that appear stuck, and that will trigger redelivery. (Circut breaker)

Because resending messages to the same consumer will likely end in not seeing the messages processed (why should the message be processed the second time if it didn't work the first time?)


Cc @michaeljmarshall
